config/confighttp/compression.go

```diff
@@ -165,13 +165,22 @@ func (d *decompressor) wrap(h http.Handler) http.Handler {
 
 func newBodyReader(r *http.Request) (io.ReadCloser, error) {
 	switch r.Header.Get("Content-Encoding") {
-	case "gzip":
+	case string(configcompression.Gzip):
```

Comment:
do you need to convert every time? Is that needed?

Reply:
I don't think these enums are supposed to represent HTTP content encodings in the abstract anyway? I'd prefer to either keep the existing strings, or add new constants specifically for the encodings, or a `GetHTTPEncoding` method on `CompressionType` which would return the string directly.